src/sssom/parsers.py

```diff
@@ -193,44 +194,35 @@ def parse_sssom_table(
     stream: io.StringIO = _open_input(file_path)
     sep_new = _get_seperator_symbol_from_file_path(file_path)
     df, sssom_metadata = _read_pandas_and_metadata(stream, sep_new)
-    # if mapping_predicates:
-    #     # Filter rows based on presence of predicate_id list provided.
-    #     df = df[df["predicate_id"].isin(mapping_predicates)]
-
-    # If SSSOM external metadata is provided, merge it with the internal metadata
-
-    if sssom_metadata:
-        if meta:
-            for k, v in meta.items():
-                if k in sssom_metadata:
-                    if sssom_metadata[k] != v:
-                        logging.warning(
-                            f"SSSOM internal metadata {k} ({sssom_metadata[k]}) "
-                            f"conflicts with provided ({meta[k]})."
-                        )
-                else:
-                    logging.info(f"Externally provided metadata {k}:{v} is added to metadata set.")
-                    sssom_metadata[k] = v
-        meta = sssom_metadata
-
-        if PREFIX_MAP_KEY in sssom_metadata:
-            if prefix_map:
-                for k, v in prefix_map.items():
-                    if k in sssom_metadata[CURIE_MAP]:
-                        if sssom_metadata[CURIE_MAP][k] != v:
-                            logging.warning(
-                                f"SSSOM prefix map {k} ({sssom_metadata[CURIE_MAP][k]}) "
-                                f"conflicts with provided ({prefix_map[k]})."
-                            )
-                    else:
-                        logging.info(
-                            f"Externally provided metadata {k}:{v} is added to metadata set."
-                        )
-                        sssom_metadata[CURIE_MAP][k] = v
-            prefix_map = sssom_metadata[CURIE_MAP]
+    if meta is None:
+        meta = {}
+
+    # The priority order for combining prefix maps are:
+    #  1. Internal prefix map inside the document
+    #  2. Prefix map passed through this function inside the ``meta``
+    #  3. Prefix map passed through this function to ``prefix_map`` (handled with ensure_converter)
+    #  4. Default prefix map (handled with ensure_converter)
+    converter = curies.chain(
+        [
+            Converter.from_prefix_map(sssom_metadata.pop(CURIE_MAP, {})),
+            Converter.from_prefix_map(meta.pop(CURIE_MAP, {})),
+            ensure_converter(prefix_map),
+        ]
+    )
+
+    # The priority order for combining metadata is:
+    #  1. Metadata appearing in the SSSOM document
+    #  2. Metadata passed through ``meta`` to this function
+    #  3. Default metadata
```

Comment:
The only downside of this is that you cannot at all _change_ the metadata in the mapping set. A common use case would be to copy a mapping set object, then updating the metadata in there with additional things, replacing, for example the contributor set. Did you check how this approaches behaves with the `sssom merge` method?

Reply:
As far as I can tell, `sssom merge` doesn't explicitly do any explicit metadata merging

Reply:
If you want me to change the priority order, then I am happy to do that. Before, it wasn't really defined in a way that made sense, so this was just my best guess. Maybe we should put the `meta` argument higher so you can override what's in the document, but does that make sense? Shouldn't we assume ahead of time that the document it internally self-consistent? If we want to update the way the data is inside, we can do a combination of reconciliation + (re-)standardization

Reply:
> As far as I can tell, sssom merge doesn't explicitly do any explicit metadata merging 

I remember now, it should not actually. Brain fart.

>  Shouldn't we assume ahead of time that the document it internally self-consistent?

I cannot currently think through all eventuality just in my mind, but intuitively I think what you have here makes sense. We will probably have to provide explicit methods to update annotations if necessary (and we already have the `sssom annotate` command so that should be covered.

I will resolve this issue for now! Thanks for indulging me!